core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -2654,6 +2654,8 @@ private RecordCursorFactory generateSelectAnalytic(QueryModel model, SqlExecutio
 
                 if (osz > 0 && !dismissOrder) {
                     IntList order = toOrderIndices(chainMetadata, ac.getOrderBy(), ac.getOrderByDirection());
+                    // set compiler for analytic
+                    analyticFunction.setRecordComparator(recordComparatorCompiler.compile(chainTypes,order));
```

Comment:
Would be nice to avoid "paying" for the comparator for the `row_number()` function and all other functions that don't need it. To achieve that, we could turn `setRecordComparator` into `initRecordComparator(RecordComparatorCompiler, IntList)` and let the function compile the comparator.